1163 - Implement Job::submit_created and Job::terminate_submitted class methods for bulk operations

nozomione opened the pull request:
## Issue Number

Closes #1163

## Purpose/Implementation Notes

I've implemented `Job::submit_created` and `Job::terminate_submitted` class methods for bulk operations, and added the corresponding tests. 

Other changes include:
- Updated the references to the `JobStatus` enum members using `.value`
- Defined `batch_job_id` using `factory.Sequence` and the default `batch_job_name` in `JobFactory`, and cleaned up `setUp` and no longer required arguments in `TestJob`

## Types of changes

- Refactor (addresses code organization and design mentioned in corresponding issue)
- New feature (non-breaking change which adds functionality)

## Functional tests
For `Job::submit_created`:
- `TestJob::test_submit_created`
- `TestJob::test_submit_created_failure`
- `TestJob::test_submit_created_no_submission`

For `Job::terminate_submitted`:
- `TestJob::test_terminate_submitted`
- `TestJob::test_terminate_submitted_failure`
- `TestJob::test_terminate_submitted_no_termination`


## Checklist

<!-- Put an `x` in the boxes that apply. -->

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

N/A

davidsmejia commented on api/scpca_portal/models/job.py:
```diff
@@ -104,36 +111,125 @@ def get_sample_job(
         )
 
+    @classmethod
+    def submit_created(cls, list_of_jobs: List[Self]) -> bool:
```
In order to get around the issue with saving or not saving the job before submitting. Let's just query `CREATED` jobs instead. This should be easier to track. It can just return a list of jobs that were submitted.

davidsmejia commented on api/scpca_portal/models/job.py:
```diff
@@ -104,36 +111,125 @@ def get_sample_job(
+        return False
+
+    @classmethod
+    def terminate_submitted(cls, retry_on_termination: bool = False) -> bool:
```
Update this to match error handling above. Also, remove logging as batch.terminate_job should handle that.

davidsmejia reviewed (commented):
Some code reduction changes requested, also can we use .value instead of .name for enums.

nozomione commented:
I've updated the following: 
- Modifired the `Job::submited_created` and `Job::terminate_submitted` class methods to return a list of updated jobs
- Used `.value` to reference `JobStatus` enum members
- Added default values for `batch_job_name` and `batch_job_id` (using `factory.Sequence`)
- Adjusted the corresponding tests for these class methods to reflect the changes

This PR is ready for another look. Thank you, David!

davidsmejia reviewed (approved):
LGTM
